keras_nlp/src/layers/modeling/reversible_embedding.py

```diff
@@ -143,20 +144,34 @@ def save_own_variables(self, store):
         if not self.built:
             return
         super().save_own_variables(store)
-        # Before Keras 3.2, the reverse weight is saved in the super() call.
-        # After Keras 3.2, the reverse weight must be saved manually.
-        if len(store.keys()) < len(self.weights):
-            # Store the reverse embedding as the last weight.
-            store[str(len(store.keys()))] = self.reverse_embeddings
+        target_variables = []
+        if not self.tie_weights:
+            # Store the reverse embedding weights as the last weights.
+            target_variables.append(self.reverse_embeddings)
+            if self.quantization_mode is not None:
+                if self.quantization_mode == "int8":
+                    target_variables.append(self.reverse_embeddings_scale)
+                else:
+                    raise self._quantization_mode_error(self.quantization_mode)
```

Comment:
do we really need to handle these modes? seems like for `save_own_variables` and `load_own_variables` we always hit build first, and build will handle these errors.

Reply:
I have simplified the condition. Fixed.